File: backend/StuMoov/StuMoov/Models/BookingModel/Booking.cs
using StuMoov.Models.BookingModel;
using System.ComponentModel.DataAnnotations;

public class Booking
{
    public Guid Id { get; private set; }
    [Required]
    public Guid PaymentId { get; private set; }
    [Required]
    public Guid RenterId { get; private set; }
    [Required]
    public Guid StorageLocationId { get; private set; }
    [Required]
    public DateTime StartDate { get; set; }
    [Required]
    public DateTime EndDate { get; set; }
    public BookingStatus Status { get; set; }
    [Required]
    public decimal TotalPrice { get; set; }

    public Booking(Guid renterId, Guid storageLocationId, DateTime startDate, DateTime endDate, decimal totalPrice)
    {
        Id = Guid.NewGuid();
        RenterId = renterId;
        StorageLocationId = storageLocationId;
        StartDate = startDate;
        EndDate = endDate;
        Status = BookingStatus.PENDING;
        TotalPrice = totalPrice;
    }
}

File: frontend/app/types/booking.ts
import { User } from "./user";
import { StorageLocation } from "./storage";
import { Payment } from "./payment";

export enum BookingStatus {
  PENDING = "PENDING",
  CONFIRMED = "CONFIRMED",
  CANCELLED = "CANCELLED",
}

export type Booking = {
  id: string;
  renterId: string;
  storageLocationId: string;
  paymentId: string | null;
  startDate: string;
  endDate: string;
  totalPrice: number | null;
  status: BookingStatus;
  createdAt: string;
  updatedAt: string;
  renter?: User;
  payment?: Payment;
  storageLocation?: StorageLocation;
};
